source/ClientApp/src/app/vendor-details/vendor-details.component.ts

```diff
@@ -1,57 +1,56 @@
 import { Component, OnInit } from '@angular/core';
 import { AuthService } from '../shared/services/auth.service';
-import { MatSnackBar } from '@angular/material';
-import { Router } from '@angular/router';
+import { Router, ActivatedRoute, ParamMap } from '@angular/router';
 import { VendorServicesService } from '../vendor-services/Services/vendor-services.service';
 import { VendorService } from '../vendors/Services/vendor.service';
 import { VendorServices } from '../shared/models/vendor-services.model';
 import { Vendor } from '../shared/models/vendor.model';
 import { Address } from '../shared/models/address.model';
+import { User } from '../shared/models/user.model';
 
 @Component({
   selector: 'app-vendor-details',
   templateUrl: './vendor-details.component.html',
   styleUrls: ['./vendor-details.component.css']
 })
 export class VendorDetailsComponent implements OnInit {
-  
+
   vendor: Vendor;
   vendorServices: VendorServices[];
   vendorId = 0;
   userName: string;
-  address: Address;
-  
+
   constructor(
     private auth: AuthService,
     private vendorServicesService: VendorServicesService,
     private vendorService: VendorService,
-    private router: Router,
+    private route: ActivatedRoute,
     ) {
   }
 
   ngOnInit() {
     if (this.auth.user) {
-      this.userName = this.auth.user.userName;
```

Comment:
If a user who is not a vendor is logged in, will this try to set the vendor to the logged in user?

Reply:
good catch